README_fr.md

```diff
@@ -0,0 +1,1643 @@
+hint: See the 'Note about fast-forwards' in 'git push --help' for details.
+```
+
+Notez que, tout comme avec la fonction `rebase` (voir ci-dessous), la fonction `amend` **remplace l'ancien commit avec un nouveau**, donc vous devez pousser de force (`--force-with-lease`) vos changements si vous avez déjà poussé le commit vers votre dépôt distant avant sa modification.
+
+```sh
+(ma-branche)$ git push origin ma-branche --force-with-lease
+```
+
+En règle générale, **évitez de pousser de force**. Il est préférable de créer et pousser un nouveau commit plutôt que de pousser de force un commit modifié car cela causera des conflits dans l'historique pour n'importe quel développeur·euse ayant déjà intéragit avec la branche en question ou quelconque branche fille. `--force-with-lease` échouera toujours, si quelqu'un d'autre était déjà en train de travailler sur la même branche que vous, et votre push réécrirait par-dessus ces modifications.
+
+Si vous êtes *absolument* sûr·e que personne n'est en train de travailler sur la même branche que vous ou que vous souhaitez mettre à jour la branche de manière *inconditionnelle*, vous pouvez utiliser `--force` (`-f`), mais cela devrait être évité en général.
+
+<a href="undo-git-reset-hard"></a>
+### J'ai fait un hard reset par accident, et je veux retrouver mes changements
+
+Si vous avez accidentellement fait un `git reset --hard`, vous pouvez normalement retrouver votre commit, car git garde un log de tout ce que vous faîtes pendant quelques jours.
+
+À noter : Cela n'est valide que si votre travail est sauvegardé, c'est à dire dans un commit ou un stash. `git reset --hard` _supprimera_ les modifications non comitées, donc utilisez cela avec précaution. (Une option plus sûre est `git reset --keep`)
+
+```sh
+(master)$ git reflog
+```
+
+Vous verrez une liste de vos précédents commits, et un commit pour la réinitialisation. Choisissez le SHA du commit vers lequel vous souhaitez retourner, et réinitialisez à nouveau :
+
+```sh
+(master)$ git reset --hard SHA1234
+```
+
+Et cela devrait faire l'affaire.
+
+<a href="undo-a-commit-merge"></a>
+### J'ai comitté et poussé une fusion par accident
+
+Si vous avez accidentellement fusionné une branche d'une fonctionnalité avec la branche de développement principale avant qu'elle ne soit prête à être fusionnée, vous pouvez toujours annuler cette fusion. Mais il y a un piège : Un commit de fusion a plus d'un parent (en général deux).
+
+La commande à utiliser
+```sh
+(feature-branch)$ git revert -m 1 <commit>
+```
+où l'option -m 1 demande de sélectionner le parent numéro 1 (la branche vers laquelle la fusion a été faite) comme parent vers lequel annuler la fusion.
+
+À noter : the parent number is not a commit identifier. Un commit de fusion ressemble plus à `Merge: 8e2ce2d 86ac2e7`. Le numéro du parent est l'index basé sur 1 du parent souhaité sur cette ligne, le premier identifiant est le numéro 1, le second le numéro 2, et ainsi de suite.
```

Comment:
Is "the parent is not a commit identifier" supposed to not be translated?
I think this bit here is a bit unclear overall, too.